### PR TITLE
INT-3936 Add support to negation for GlobalChannelInterceptor and use common utility method for IntegrationManagementConfigurer and IntegrationMBeanExporter

### PR DESCRIPTION
**Pull request by meherzad**
INT-3936 Add support to negation for GlobalChannelInterceptor and use common utility method for IntegrationManagementConfigurer and IntegrationMBeanExporter

JIRA: https://jira.spring.io/browse/INT-3936

Add support for negation for GlobalChannelInterceptor. Create common utility method to be used by other classes.

### CODE REVIEW

File: spring-integration-core/src/main/java/org/springframework/integration/config/GlobalChannelInterceptor.java [hunk 1]
```diff
@@ -41,8 +41,8 @@
 public @interface GlobalChannelInterceptor {
 
 	/**
-	 * An array of simple patterns against which channel names will be matched. Default is "*"
-	 * (all channels). See {@link org.springframework.util.PatternMatchUtils#simpleMatch(String, String)}.
+	 * An array of patterns against which channel names will be matched. Default is "*"
+	 * (all channels). See {@link org.springframework.integration.util.PatternMatchUtils#smartMatch(String[], String)}.
```
Comment (artembilan) [inline spring-integration-core/src/main/java/org/springframework/integration/config/GlobalChannelInterceptor.java]: That `See` can move to the `@see` JavaDoc tag after `@return`.
Plus that would be great to mention in the general description a negation ability.

**Review by artembilan — CHANGES_REQUESTED** [review]
That's great work, @meherzad !
Thank you!

But still need some improvements. 
See comments on lines and consider these:

* In the `spring-integration-5.0.xsd` the `<xsd:element name="channel-interceptor">` in its `pattern` attribute should mention that negative patterns are supported as well. Plus I would like to see some negative test for XML support as well. E.g. in the `GlobalChannelInterceptorTests` and its config nearby.
* Since we make this change and it is definitely and improvement would be great to mention this news in the `whats-new.adoc` with the link for more detailed description in the `channel.adoc` under the `[[global-channel-configuration-interceptors]]` chapter.
* In the future consider to make a header of the first comment not more than 50 symbols. Then blank line. Then link to JIRA and after that detailed description. You can find sample in the commit I did for your previous contribution: https://github.com/spring-projects/spring-integration/blob/master/CONTRIBUTING.adoc#provide-a-link-to-the-jira-issue-in-the-associated-pull-request. Meanwhile the whole bunch of rules how to write the proper commit message here: https://chris.beams.io/posts/git-commit/

We have time until tomorrow though if you would like to include this into the current M7. Otherwise the next one. Although I think you just contribute as is without any particular aim 😄 . Thank you very much for your effort!

File: spring-integration-core/src/main/java/org/springframework/integration/util/PatternMatchUtils.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.util;
+
+/**
+ * Utility methods for pattern matching.
+ * @author Meherzad Lahewala
+ * @since 5.0
```
[the remark this answers is not in the document]
Comment (meherzad) [reply spring-integration-core/src/main/java/org/springframework/integration/util/PatternMatchUtils.java]: What you suggested does make sense. Will fix the method and add required doc for the same.

**Comment by meherzad** [comment]
@artembelin - All the review comments have been addressed.

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration/commit/40783ff54762b69d4f403c20b98cc69aa3786cc8.

@meherzad ,

thank you for contribution and, of course, we are looking forward for more in the future!

One small remark: consider do not squash commits in the PR - that will be easier for review.

**Pull request closed without merging**